bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -1005,7 +1005,10 @@ impl TimelineItem {
 #[derive(Clone, uniffi::Enum)]
 pub enum EventSendState {
     /// The local event has not been sent yet.
-    NotSentYet,
+    NotSentYet {
+        /// When the send was first enqueued by the user.
+        created_at: Option<u64>,
```

Comment:
Random thought: it seems to me that the `created_at` property is not so much depending on the send state, but it's useful for all local echoes in general. How about putting it in `LocalEventTimelineItem`, instead? That might get rid of many of the test changes (and cause many changes to be reverted, sorry!).

Reply:
Note a problem at all, you are correct, this is much cleaner, and makes for a much smaller change.

I was too focused on trying to get it into the NotSentYet state, as opposed to exposing it as a parallel field on the TimelineItem in the FFI.